### PR TITLE
Update dev

### PR DESCRIPTION
**Pull request by nikk-nikaznan**
Port to lightning

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,21 @@
+MIT License
+
+Copyright (c) 2020 Amir Atapour-Abarghouei
```
Comment (sbonner0) [inline LICENSE]: Not sure why this person gets the copyright?

File: src/EEG_1SCU_CM.py [hunk 2]
```diff
@@ -1,177 +1,182 @@
...
+        input_data, input_label, config["batch_size"], args.seed_n
+    )
+    model = SCU_Model(config)
+
+    trainer = pl.Trainer(
+        max_epochs=config["num_epochs"],
+        logger=True,
+        log_every_n_steps=1,
+        accelerator="cpu",
+    )
```
Comment (sbonner0) [inline src/EEG_1SCU_CM.py]: you could allow some more options to be set here including setting mixed precision etc

File: src/EEG_1SCU_CM.py [hunk 3]
```diff
@@ -1,177 +1,182 @@
...
+            batch (Tuple[torch.Tensor, torch.Tensor]): A tuple containing input data and corresponding labels.
+            batch_idx (int): Index of the batch.
+        """
+        x, y = batch
+        y_hat = self.model(x)
+        test_loss = self.criterion(y_hat, y.long())
+        test_acc = self.test_acc(torch.argmax(y_hat, dim=1), y.long())
+        self.test_step_outputs.append(
+            {"test_loss": test_loss, "test_acc": test_acc, "y": y, "y_hat": y_hat}
+        )
```
Comment (sbonner0) [inline src/EEG_1SCU_CM.py:111]: Might be worth revising this as I think this the test_step should return something?

File: src/datamodule.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        Setup datasets for training and testing.
+
+        Args:
+            stage (str, optional): Stage of setup (train, test).
+            Defaults to None.
+        """
+        generator = torch.Generator().manual_seed(self.seed_n)
+        # Convert your input data and labels to PyTorch tensors
+        input_data_tensor = torch.tensor(self.input_data, dtype=torch.float64)
+        input_label_tensor = torch.tensor(self.input_label, dtype=torch.float64)
```
Comment (sbonner0) [inline src/datamodule.py]: these should probably be float 32

File: src/datamodule.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        )
+
+    def train_dataloader(self) -> DataLoader:
+        """
+        Returns a DataLoader for the training set.
+
+        Returns:
+            DataLoader: DataLoader for the training set.
+        """
+        return DataLoader(self.train_dataset, batch_size=self.batch_size, shuffle=True)
```
Comment (sbonner0) [inline src/datamodule.py:62]: do you not want shuffle on here?

**Review by sbonner0 — COMMENTED** [review]
Just a few small comments @nikk-nikaznan

File: src/EEG_1SCU_CM.py (hunk 2 above)
Comment (nikk-nikaznan) [reply src/EEG_1SCU_CM.py]: Will add issue for that
